src/main/java/de/tum/in/www1/exerciseapp/web/rest/ResultResource.java

```diff
@@ -195,7 +196,18 @@ public ResultResource(ResultRepository resultRepository, Optional<LtiService> lt
         } else {
             results = resultRepository.findEarliestSuccessfulResultsForExercise(exerciseId);
         }
-        return results;
+
+        List<ResultWithSubmissionCount> resultsWithSubmissionCounts = new ArrayList<>();
+        List<Object[]> submissionCounts = resultRepository.findSubmissionCountsForStudents(exerciseId);
```

Comment:
Can you add some comments here? Why do we e.g. have List<Object[]>? This looks a bit weired.